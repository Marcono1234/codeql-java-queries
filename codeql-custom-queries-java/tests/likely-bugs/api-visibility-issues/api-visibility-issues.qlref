./likely-bugs/api-visibility-issues.ql
