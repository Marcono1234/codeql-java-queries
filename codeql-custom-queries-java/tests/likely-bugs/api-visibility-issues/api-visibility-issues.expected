| mypackage/PublicClass.java:6:14:6:24 | PublicClass | Exposes type $@ with lower module visibility 'public' through $@ with module visibility 'to all modules' | notexportedpackage/NotExportedClass.java:9:14:9:29 | NotExportedClass | NotExportedClass | mypackage/PublicClass.java:8:29:8:42 | getNotExported | return type of method getNotExported() |
| mypackage/PublicClass.java:6:14:6:24 | PublicClass | Exposes type $@ with lower module visibility 'to modules java.base' through $@ with module visibility 'to all modules' | restrictedpackage/RestrictedClass.java:8:14:8:28 | RestrictedClass | RestrictedClass | mypackage/PublicClass.java:13:28:13:45 | getRestrictedClass | return type of method getRestrictedClass() |
| mypackage/PublicClass.java:6:14:6:24 | PublicClass | Exposes type $@ with lower visibility 'package-private' through $@ with effective visibility 'public' | mypackage/PackagePrivateClass.java:3:7:3:25 | PackagePrivateClass | PackagePrivateClass | mypackage/PublicClass.java:18:32:18:53 | getPackagePrivateClass | return type of method getPackagePrivateClass() |
| mypackage/PublicClass.java:6:14:6:24 | PublicClass | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/PublicClass.java:22:26:22:37 | PrivateClass | PrivateClass | mypackage/PublicClass.java:26:25:26:39 | getPrivateClass | return type of method getPrivateClass() |
| mypackage/PublicClass.java:6:14:6:24 | PublicClass | Exposes type $@ with lower visibility 'protected (declared by PublicClass)' through $@ with effective visibility 'public' | mypackage/PublicClass.java:30:28:30:41 | ProtectedClass | ProtectedClass | mypackage/PublicClass.java:34:27:34:43 | getProtectedClass | return type of method getProtectedClass() |
| mypackage/PublicClass.java:48:25:48:42 | DifferentProtected | Exposes type $@ with lower visibility 'protected (declared by PublicClass)' through $@ with effective visibility 'protected (declared by DifferentProtected)' | mypackage/PublicClass.java:30:28:30:41 | ProtectedClass | ProtectedClass | mypackage/PublicClass.java:50:34:50:50 | getProtectedClass | return type of method getProtectedClass() |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:14:25:14:37 | exposingField | type of field 'exposingField' |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:25:13:25:34 | T | type variable 'T' of constructor Test(PrivateClass) |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:25:42:25:55 | c | parameter 'c' of constructor Test(PrivateClass) |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:25:65:25:76 | PrivateClass | thrown exception of constructor Test(PrivateClass) |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:35:13:35:34 | T | type variable 'T' of method exposingMethod(PrivateClass, T) |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:35:50:35:63 | exposingMethod | return type of method exposingMethod(PrivateClass, T) |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:35:65:35:79 | p1 | parameter 'p1' of method exposingMethod(PrivateClass, T) |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:35:95:35:106 | PrivateClass | thrown exception of method exposingMethod(PrivateClass, T) |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:72:20:72:52 | T | type variable 'T' of method typeExposing(List<? extends PrivateClass>, List<? super PrivateClass>, PrivateClass[], PrivateClass[][], List<? extends List<? super PrivateClass>[]>) |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:72:73:72:103 | p1 | parameter 'p1' of method typeExposing(List<? extends PrivateClass>, List<? super PrivateClass>, PrivateClass[], PrivateClass[][], List<? extends List<? super PrivateClass>[]>) |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:72:106:72:134 | p2 | parameter 'p2' of method typeExposing(List<? extends PrivateClass>, List<? super PrivateClass>, PrivateClass[], PrivateClass[][], List<? extends List<? super PrivateClass>[]>) |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:72:137:72:153 | p3 | parameter 'p3' of method typeExposing(List<? extends PrivateClass>, List<? super PrivateClass>, PrivateClass[], PrivateClass[][], List<? extends List<? super PrivateClass>[]>) |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:72:156:72:174 | p4 | parameter 'p4' of method typeExposing(List<? extends PrivateClass>, List<? super PrivateClass>, PrivateClass[], PrivateClass[][], List<? extends List<? super PrivateClass>[]>) |
| mypackage/Test.java:5:14:5:17 | Test | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:72:177:72:223 | p5 | parameter 'p5' of method typeExposing(List<? extends PrivateClass>, List<? super PrivateClass>, PrivateClass[], PrivateClass[][], List<? extends List<? super PrivateClass>[]>) |
| mypackage/Test.java:10:25:10:39 | ExposingGeneric | Type with visibility 'public' exposes type $@ with lower visibility 'private' through $@ | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:10:41:10:62 | T | type variable 'T' |
| mypackage/Test.java:39:25:39:39 | InheritanceBase | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:40:29:40:41 | exposingField | type of field 'exposingField' |
| mypackage/Test.java:39:25:39:39 | InheritanceBase | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:42:29:42:42 | exposingMethod | return type of method exposingMethod() |
| mypackage/Test.java:39:25:39:39 | InheritanceBase | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:46:29:46:47 | otherExposingMethod | return type of method otherExposingMethod() |
| mypackage/Test.java:52:29:52:42 | NestedExposing | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:53:33:53:46 | exposingMethod | return type of method exposingMethod() |
| mypackage/Test.java:60:25:60:42 | InheritingExposing | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:40:29:40:41 | exposingField | type of inherited field InheritanceBase.exposingField |
| mypackage/Test.java:60:25:60:42 | InheritingExposing | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:42:29:42:42 | exposingMethod | return type of inherited method InheritanceBase.exposingMethod() |
| mypackage/Test.java:85:25:85:51 | TransitivelyGenericExposing | Exposes type $@ with lower visibility 'private' through $@ with effective visibility 'public' | mypackage/Test.java:6:26:6:37 | PrivateClass | PrivateClass | mypackage/Test.java:76:18:76:20 | get | return type of inherited method Generic.get() |
| otherrestrictedpackage/OtherRestrictedClass.java:5:14:5:33 | OtherRestrictedClass | Exposes type $@ with lower module visibility 'to modules java.base' through $@ with module visibility 'to modules java.compiler' | restrictedpackage/RestrictedClass.java:8:14:8:28 | RestrictedClass | RestrictedClass | otherrestrictedpackage/OtherRestrictedClass.java:7:28:7:45 | getRestrictedClass | return type of method getRestrictedClass() |
| restrictedpackage/RestrictedClass.java:8:14:8:28 | RestrictedClass | Exposes type $@ with lower visibility 'package-private' through $@ with effective visibility 'public' | restrictedpackage/PackagePrivateClass.java:3:7:3:25 | PackagePrivateClass | PackagePrivateClass | restrictedpackage/RestrictedClass.java:15:32:15:53 | getPackagePrivateClass | return type of method getPackagePrivateClass() |
